server/public/model/packet_metadata.go

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package model
+
+import (
+	"fmt"
+
+	"github.com/blang/semver/v4"
+	"gopkg.in/yaml.v2"
+)
+
+type PacketType string
+
+const (
+	CurrentMetadataVersion int        = 1
+	SupportPacketType      PacketType = "support-packet"
+	PluginPacketType       PacketType = "plugin-packet"
```

Comment:
@ifoukarakis I wonder if plugins should differentiate their metadata using customer Packet types. For example, the metrics plugin would use `metrics-packet`, while the NPS one uses `nps-packet` as the `Type`. What do you think?

Reply:
The `type` field  is mostly useful in order to know what to expect in `extras`. The goals of introducing `type` and plugin id/version in extras were:
- Be able to distinguish on the process generating the current packet.
- Set expectation on the schema of `extras`.

There are a few options that have been discussed:

1. Use two distinct values, indicating if it's a support packet generated by MM server or a plugin. In case of plugin, the plugin id and version should be part of `extras` in order to handle schema changes over versions. In this case, an "enum" with two values (like in this PR) can be used.
2. A variation of this would be to move plugin id and version to top level (rather than extras). MM server can define a special reserved "plugin id", although the server is not an ID. The combination of plugin id/version must be unique, helping map to a distinct schema for `extras`. We can expect In this case I believe we can remove `type`.
3. Use an enum with a special value for each plugin. The main drawback is that the server will need to keep all the different values for each plugin, coupling plugins generating packages with server code.

The previous PR was suggesting the option #1. Your suggestion is closer to option #2. I have no strong preference between the two options. Perhaps @isacikgoz can add another point of view.

Reply:
My point was making the code independent so that whenever a new plugin is added, we wouldn't require adding a new type there. I think the biggest differentiator is the server vs plugin and it is good as it is to me.

Reply:
I wonder if we even need to make the server code depend on the on newer plugins. A plugin can simply provide it's own `type` though the API. (This is #3)

Reply:
Please keep in mind this metadata needs to be generated by new plugins on existing ESRs as well, so any sort of dependency on MM server's code won't work. This has to be independent of MM server.

Reply:
Sure thing, @harshilsharma63; supporting the ESR version of MM was one of the motivations behind moving the generation code to a library method.

Reply:
Let's keep things as they are. The plugin_id is fine for separating the different packet types. We can still add new types later.